src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,998 @@
+				// TODO: routing control for screen destination
+				break;
+			case SUPER_AUX_DESTINATION:
+				controlAssignSourceToSuperAuxDest(value, propertyValue);
+				break;
+			default:
+				if (logger.isWarnEnabled()) {
+					logger.warn(String.format("Operation %s with value %s is not supported.", property, value));
+				}
+				throw new IllegalArgumentException(String.format("Operation %s with value %s is not supported.", property, value));
+		}
+	}
+
+	/**
+	 * controlProperty: Assign source to Super Aux Destination
+	 *
+	 * @param destName Destination name that will be assigned a new source to
+	 * @param sourceName name of the new source
+	 */
+	private void controlAssignSourceToSuperAuxDest(String sourceName, String destName) {
+		String auxDestName = getAuxDestinationName(destName);
+		boolean changeSuperAuxResult = changeAuxContent(auxDestName, sourceName);
+		if (logger.isDebugEnabled()) {
+			String debugString = changeSuperAuxResult ? String.format("Assign source: %s to Super Aux Destination %s success!", destName, sourceName)
+					: String.format("Assign source: %s to Super Aux Destination %s fail!", destName, sourceName);
+			logger.debug(debugString);
+		}
+	}
+
+	/**
+	 * controlProperty: Assign source to Aux Destination
+	 *
+	 * @param destName Destination name that will be assigned a new source to
+	 * @param sourceName name of the new source
+	 */
+	private void controlAssignSourceToAuxDest(String destName, String sourceName) {
+		boolean changeAuxResult = changeAuxContent(sourceName, destName);
+		if (logger.isDebugEnabled()) {
+			String debugString = changeAuxResult ? String.format("Assign source: %s to Aux Destination %s success!", destName, sourceName)
+					: String.format("Assign source: %s to Aux Destination %s fail!", destName, sourceName);
+			logger.debug(debugString);
+		}
+	}
+
+	/**
+	 * controlProperty: Activate a preset.
+	 */
+	private void controlActivatePreset() {
+		boolean result;
+		result = activatePresetResult();
+		if (logger.isDebugEnabled()) {
+			String debugString = result ? String.format("Activate %s success!", lastPresetName) : String.format("Activate %s fail!", lastPresetName);
+			logger.debug(debugString);
+		}
+	}
+
+	/**
+	 * Control properties
+	 *
+	 * @param list list of control properties that will be controlled
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) {
+		if (CollectionUtils.isEmpty(list)) {
+			throw new IllegalArgumentException("Controllable properties cannot be null or empty");
+		}
+		for (ControllableProperty controllableProperty : list) {
+			controlProperty(controllableProperty);
+		}
+	}
+
+	/**
+	 * Get Aux destination name from SuperAux-AuxDestination String
+	 *
+	 * @param propertyValue String SuperAuxName-AuxName
+	 */
+	private String getAuxDestinationName(String propertyValue) {
+		try {
+			String[] splitString = propertyValue.split(BarcoE2Constant.DASH);
+			return splitString[1];
```

Comment:
Yes, the input string should be "SuperAuxName-AuxName" so there are only 2 items in the array.

Reply:
Ok I see. I think it's fine here anyways since any error (such as index out of bounds) will be caught and rethrown with the correct message. But I would ask you to reconsider usage of  ResourceConflictException, maybe IllegalArgumentException would be a better fit here?